### PR TITLE
Non-controversial part of #44523

### PR DESCRIPTION
**Pull request by caesarxuchao**
For easier review of #44523, i extracted the non-controversial part out to this PR.

### CODE REVIEW

File: staging/src/k8s.io/apimachinery/pkg/api/meta/help.go [hunk 1]
```diff
@@ -175,6 +175,9 @@ func SetList(list runtime.Object, objects []runtime.Object) error {
 	slice := reflect.MakeSlice(items.Type(), len(objects), len(objects))
 	for i := range objects {
 		dest := slice.Index(i)
+		if dest.Type() == reflect.TypeOf(runtime.RawExtension{}) {
```
Comment (caesarxuchao) [inline staging/src/k8s.io/apimachinery/pkg/api/meta/help.go]: Explained here: https://github.com/kubernetes/kubernetes/pull/44523#discussion_r113587700

File: staging/src/k8s.io/client-go/dynamic/client_pool.go [hunk 2]
```diff
@@ -73,9 +72,9 @@ func NewClientPool(config *restclient.Config, mapper meta.RESTMapper, apiPathRes
 
 // Instantiates a new dynamic client pool with the given config.
 func NewDynamicClientPool(cfg *restclient.Config) ClientPool {
-	// TODO: should use a dynamic RESTMapper built from the discovery results.
-	restMapper := api.Registry.RESTMapper()
-	return NewClientPool(cfg, restMapper, LegacyAPIPathResolverFunc)
+	// restMapper is not needed when using LegacyAPIPathResolverFunc
```
Comment (caesarxuchao) [inline staging/src/k8s.io/client-go/dynamic/client_pool.go]: @lavalamp this is a manual change.

File: staging/src/k8s.io/client-go/kubernetes/typed/core/v1/pod_expansion.go [hunk 3]
```diff
@@ -41,5 +41,5 @@ func (c *pods) Evict(eviction *policy.Eviction) error {
 
 // Get constructs a request for getting the logs for a pod
 func (c *pods) GetLogs(name string, opts *v1.PodLogOptions) *restclient.Request {
-	return c.client.Get().Namespace(c.ns).Name(name).Resource("pods").SubResource("log").VersionedParams(opts, api.ParameterCodec)
+	return c.client.Get().Namespace(c.ns).Name(name).Resource("pods").SubResource("log").VersionedParams(opts, scheme.ParameterCodec)
```
Comment (caesarxuchao) [inline staging/src/k8s.io/client-go/kubernetes/typed/core/v1/pod_expansion.go]: Explained here: https://github.com/kubernetes/kubernetes/pull/44523#discussion_r113590819

File: pkg/api/v1/resource/helpers.go [hunk 4]
```diff
@@ -112,15 +111,19 @@ func ExtractResourceValueByContainerName(fs *v1.ResourceFieldSelector, pod *v1.P
...
 }
 
+type deepCopier interface {
+	DeepCopy(interface{}) (interface{}, error)
+}
+
 // ExtractResourceValueByContainerNameAndNodeAllocatable extracts the value of a resource
 // by providing container name and node allocatable
-func ExtractResourceValueByContainerNameAndNodeAllocatable(fs *v1.ResourceFieldSelector, pod *v1.Pod, containerName string, nodeAllocatable v1.ResourceList) (string, error) {
+func ExtractResourceValueByContainerNameAndNodeAllocatable(copier deepCopier, fs *v1.ResourceFieldSelector, pod *v1.Pod, containerName string, nodeAllocatable v1.ResourceList) (string, error) {
```
Comment (caesarxuchao) [inline pkg/api/v1/resource/helpers.go]: Addressed the comment

File: pkg/api/v1/types.go [hunk 5]
```diff
@@ -4428,3 +4428,37 @@ type NodeResources struct {
 	// Capacity represents the available resources of a node
 	Capacity ResourceList `protobuf:"bytes,1,rep,name=capacity,casttype=ResourceList,castkey=ResourceName"`
 }
+
+const (
```
Comment (caesarxuchao) [inline pkg/api/v1/types.go]: Not sure if it is worth creating a file. types.go is filled with many constants already.

File: staging/src/k8s.io/client-go/kubernetes/typed/core/v1/pod_expansion.go (hunk 3 above)
Comment (lavalamp) [inline staging/src/k8s.io/client-go/kubernetes/typed/core/v1/pod_expansion.go]: We need to fix this, but it doesn't need to be in this PR.

**Pull request merged**